sdk/servicebus/Azure.Messaging.ServiceBus/src/Amqp/AmqpConsumer.cs

```diff
@@ -336,6 +418,486 @@ private void CloseLink(ReceivingAmqpLink link)
+                timeout.CalculateRemaining(stopWatch.Elapsed)))
+                .ConfigureAwait(false);
+            cancellationToken.ThrowIfCancellationRequested<TaskCanceledException>();
+
+            using AmqpMessage responseAmqpMessage = await link.RequestAsync(
+                amqpRequestMessage.AmqpMessage,
+                timeout.CalculateRemaining(stopWatch.Elapsed))
+                .ConfigureAwait(false);
+
+            cancellationToken.ThrowIfCancellationRequested<TaskCanceledException>();
+
+            AmqpResponseMessage amqpResponseMessage = AmqpResponseMessage.CreateResponse(responseAmqpMessage);
+
+            var messages = new List<ServiceBusReceivedMessage>();
+            //AmqpError.ThrowIfErrorResponse(responseAmqpMessage, EntityName);
+            if (amqpResponseMessage.StatusCode == AmqpResponseStatusCode.OK)
+            {
+                ServiceBusReceivedMessage message = null;
+                IEnumerable<AmqpMap> messageList = amqpResponseMessage.GetListValue<AmqpMap>(ManagementConstants.Properties.Messages);
+                foreach (AmqpMap entry in messageList)
+                {
+                    var payload = (ArraySegment<byte>)entry[ManagementConstants.Properties.Message];
+                    var amqpMessage = AmqpMessage.CreateAmqpStreamMessage(new BufferListStream(new[] { payload }), true);
+                    message = AmqpMessageConverter.AmqpMessageToSBMessage(amqpMessage, true);
+                    messages.Add(message);
+                }
+
+                if (message != null)
+                {
+                    LastPeekedSequenceNumber = message.SequenceNumber;
+                }
+                return messages;
+            }
+
+            if (amqpResponseMessage.StatusCode == AmqpResponseStatusCode.NoContent ||
+                (amqpResponseMessage.StatusCode == AmqpResponseStatusCode.NotFound && Equals(AmqpClientConstants.MessageNotFoundError, amqpResponseMessage.GetResponseErrorCondition())))
+            {
+                return messages;
+            }
+            // TODO throw correct exception
+            throw new Exception();
+        }
+
+        /// <summary>
+        /// Updates the disposition status of deferred messages.
+        /// </summary>
+        ///
+        /// <param name="lockTokens">Message lock tokens to update disposition status.</param>
+        /// <param name="timeout"></param>
+        /// <param name="dispositionStatus"></param>
+        /// <param name="isSessionReceiver"></param>
+        /// <param name="sessionId"></param>
+        /// <param name="propertiesToModify"></param>
+        /// <param name="deadLetterReason"></param>
+        /// <param name="deadLetterDescription"></param>
+        private async Task DisposeMessageRequestResponseAsync(
+            Guid[] lockTokens,
+            TimeSpan timeout,
+            DispositionStatus dispositionStatus,
+            bool isSessionReceiver,
+            string sessionId = null,
```

Comment:
I believe you've removed these params in other PR and you need to pull latest from the master.